baseapp/server.go

```diff
@@ -130,6 +139,37 @@ func (s *Server) Start() error {
 	return s.server.ListenAndServe()
 }
 
+// Start starts the server and blocks.
+func (s *Server) Start() error {
+	// maintain backwards compatibility
+	if s.config.ShutdownWaitTime == nil {
+		return s.start()
+	}
+
+	go func() {
+		if err := s.start(); err != nil {
+			s.quit <- err
+		}
+	}()
+
+	// SIGKILL and SIGSTOP cannot be caught, so don't bother adding them here
+	interrupt := make(chan os.Signal, 2)
+	signal.Notify(interrupt, syscall.SIGINT, syscall.SIGTERM)
+
+	select {
+	case <-interrupt:
+		s.logger.Info().Msg("Caught interrupt, gracefully shutting down")
+	case err := <-s.quit:
+		if err != http.ErrServerClosed {
+			s.logger.Error().Err(err).Msg("Shutting down server")
```

Comment:
I think this error can only come from starting the server, in which case we don't need to wait for shutdown. Can we `return err` here instead?